app/javascript/courses/review/components/CoursesReview__Editor.res

```diff
@@ -1128,6 +1204,42 @@ let make = (
           <div className="p-4 md:p-6">
             <SubmissionChecklistShow checklist=state.checklist updateChecklistCB pending />
           </div>
+          {switch submissionReport {
+          | Some(report) =>
+            <div className="p-4 md:p-6 space-y-8">
+              <div className="bg-gray-300 p-4 rounded-md">
+                <div className="flex items-center justify-between text-sm">
+                  <div className="flex items-center space-x-3">
+                    <Icon className={reportStatusIconClasses(SubmissionReport.status(report))} />
+                    <p className="font-semibold">
+                      {str(reportStatusString(SubmissionReport.status(report)))}
+                    </p>
+                  </div>
+                  <button
+                    onClick={_ => send(ChangeReportVisibility)}
+                    className="inline-flex items-center text-primary-500 px-3 py-2 rounded font-semibold hover:text-primary-700 hover:bg-gray-400 focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 transition">
+                    <span className="hidden md:block pr-3">
+                      {str(
+                        state.showReport
+                          ? t("hide_test_report_button")
+                          : t("show_test_report_button"),
+                      )}
+                    </span>
+                    <Icon className="if i-arrows-expand-light text-xl" />
+                  </button>
+                </div>
+                {state.showReport
+                  ? <div>
+                      <p className="text-sm font-semibold mt-4"> {str("Test Report")} </p>
+                      <div className="bg-white p-3 rounded-md border mt-2">
+                        {str(SubmissionReport.description(report))}
```

Comment:
The test report should be treated as Markdown, and so rendered using a `MarkdownBlock`.